psi4/src/read_options.cc

```diff
@@ -1544,6 +1544,30 @@ int read_options(const std::string &name, Options &options, bool suppress_printi
         /*- Bump function max radius -*/
         options.add_double("DF_BUMP_R1", 0.0);
 
+        /*- SUBSECTION COSX Algorithm -*/
+
+        /*- Number of spherical points in initial COSX grid. -*/
+        options.add_int("COSX_SPHERICAL_POINTS_INITIAL", 50);
+        /*- Number of radial points in initial COSX grid. -*/
+        options.add_int("COSX_RADIAL_POINTS_INITIAL", 25);
+        /*- Number of spherical points in final COSX grid. -*/
+        options.add_int("COSX_SPHERICAL_POINTS_FINAL", 110);
+        /*- Number of radial points in final COSX grid. -*/
+        options.add_int("COSX_RADIAL_POINTS_FINAL", 35);
+        /*- Screening criteria for integrals and intermediates in COSX -*/
+        options.add_double("COSX_INTS_TOLERANCE", 1.0E-11);
+        /*- Screening criteria for shell-pair densities in COSX !expert -*/
+        options.add_double("COSX_DENSITY_TOLERANCE", 1.0E-10);
+        /*- Screening criteria for basis function values on COSX grids !expert -*/
+        options.add_double("COSX_BASIS_TOLERANCE", 1.0E-10);
+        /*- Pruning scheme for COSX grids !expert -*/
+        options.add_str("COSX_PRUNING_SCHEME", "ROBUST", 
+                        "ROBUST TREUTLER NONE FLAT P_GAUSSIAN D_GAUSSIAN P_SLATER D_SLATER LOG_GAUSSIAN LOG_SLATER NONE");
```

Comment:
This looks like duplication of pruning schemes. They should be collected under one variable in `read_options` and used for all instances of grid pruning choices.

Reply:
The duplication here is intentional. 

One alternative would be to determine the COSX pruning scheme from the value of the existing `DFT_PRUNING_SCHEME` keyword. I don't think that would be a good idea because (1) it's unintuitive to expect a keyword named `DFT_PRUNING_SCHEME` to affect a COSX calculation--DFT is independent of COSX--and (2) users may very likely want different DFT and COSX grids.

In the future, we could address point (1) by collecting the three keywords `DFT_PRUNING_SCHEME`, `COSX_PRUNING_SCHEME`, and `MBIS_PRUNING_SCHEME` into a `GRID_PRUNING_SCHEME` keyword as you said, but that leaves point (2) unaddressed. Also, that change would be well outside of the scope of this PR.

Reply:
That is not what I meant.

When one adds a new pruning scheme, one needs to remember to add it in every single instance of the options for DFT, MBIS, and COSX.

My suggestion was to collect the possible grid choices into a variable, e.g. `std::string grid_schemes = "ROBUST TREUTLER NONE FLAT P_GAUSSIAN D_GAUSSIAN P_SLATER D_SLATER LOG_GAUSSIAN LOG_SLATER NONE";` and assigning this variable to the three above mentioned parameters. (I am assuming that the code that parses the grid options is the same for the three algorithms.)

Reply:
Oh, I see. Yeah, that would make it cleaner to add a new grid option. @loriab would a `grid_schemes` string disrupt the autodocs? If not, I'll make the change

Reply:
I regret that this sensible suggestion will very likely flummox the autodocs. maybe add it as a todo to read_options.cc .